### PR TITLE
Added data-* and aria-* attributes

### PR DESCRIPTION
**Pull request by rmarren1**
https://github.com/plotly/dash/pull/237

### CODE REVIEW

File: scripts/generate-components.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ function generatePropTypes(element, attributes) {
      */
     'children': PropTypes.node,
 
+    /**
+     * A wild-card data-* attribute
+     */
+    'data-*': PropTypes.string,
+
```
Comment (chriddyp) [inline scripts/generate-components.js]: Let's add `aria-*` too

**Comment by pope1ni** [comment]
Hi @rmarren1,

I've also not been able to use the `role` attribute with `html.A()`:
```
TypeError: Unexpected keyword argument `role`
```

This attribute is required for use with `aria-*` attributes, so it would be good if it were supported.

**Comment by rmarren1** [comment]
Are there any elements where `role` should not be allowed? It is does not appear in https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes which is why it is currently not supported, but it could be hard-coded into the generation script if it applies to all components.

**Comment by pope1ni** [comment]
Can be applied to any element:

- https://www.w3.org/TR/html5/dom.html#aria-role-attribute

Other documents of interest:

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA
- https://www.w3.org/TR/role-attribute/
- https://www.w3.org/TR/wai-aria-1.1/
- https://www.w3.org/TR/html-aria/

Hard-coding this seems sensible if that is already being done for `aria-*` as part of this PR.

For background, I bumped into this because of using Bootstrap which requires `data-*` for some of the scripted components. Have worked around the issue in the meantime client-side by using a mutation observer and adding the attributes as the elements appear.

**Comment by rmarren1** [comment]
added `role`

**Comment by pope1ni** [comment]
Thanks!

**Comment by chriddyp** [comment]
@rmarren1 - I'm 👍 with these changes. Is there anything else that you would like to add? Otherwise, I will merge and release.

**Comment by rmarren1** [comment]
I'm good with it, go ahead!

**Pull request merged**